petclinic-frontend/src/features/customers/components/AdminUpdateCustomerForm.tsx

```diff
@@ -0,0 +1,144 @@
+import { useEffect, useState, FC } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { getOwner, updateOwner } from '@/features/customers/api/updateOwner.ts';
+import { OwnerRequestModel } from '../models/OwnerRequestModel';
+import { OwnerResponseModel } from '../models/OwnerResponseModel';
+import './UpdateCustomerForm.css';
+
+const AdminUpdateCustomerForm: FC = () => {
+  const { ownerId } = useParams<{ ownerId: string }>();
+  const navigate = useNavigate();
+  const [formData, setFormData] = useState<OwnerRequestModel>({
+    firstName: '',
+    lastName: '',
+    address: '',
+    city: '',
+    province: '',
+    telephone: '',
+  });
+
+  const [errors, setErrors] = useState<{ [key: string]: string }>({});
+
+  useEffect(() => {
+    const fetchOwnerData = async (): Promise<void> => {
+      if (!ownerId) {
+        console.error('Owner id is undefined');
+        return;
+      }
+
+      try {
+        const response = await getOwner(ownerId);
+        const ownerData: OwnerResponseModel = response.data;
+        setFormData(ownerData);
+      } catch (error) {
+        console.error('Error fetching owner data:', error);
+      }
+    };
+
+    fetchOwnerData().catch(error =>
+      console.error('Error in fetchOwnerData:', error)
+    );
+  }, [ownerId]);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    setFormData({
+      ...formData,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const validate = (): boolean => {
+    const newErrors: { [key: string]: string } = {};
+    if (!formData.firstName) newErrors.firstName = 'First name is required';
+    if (!formData.lastName) newErrors.lastName = 'Last name is required';
+    if (!formData.address) newErrors.address = 'Address is required';
+    if (!formData.city) newErrors.city = 'City is required';
+    if (!formData.province) newErrors.province = 'Province is required';
+    if (!formData.telephone) newErrors.telephone = 'Telephone is required';
+    setErrors(newErrors);
+    return Object.keys(newErrors).length === 0;
+  };
+
+  const handleSubmit = async (
+    e: React.FormEvent<HTMLFormElement>
+  ): Promise<void> => {
+    e.preventDefault();
+    if (!validate()) return;
+
+    try {
+      if (!ownerId) {
+        console.error('Owner id is undefined');
+        return;
+      }
+      await updateOwner(ownerId, formData);
```

Comment:
When updating the customer, there should be some kind of visual validation for the user to see that they succesfully updated the owner details

Reply:
You're right, adding a visual validation would make the user experience more enjoyable.